packages/web/src/pages/sign-up-page/components/ExternalWalletSignUpModal.tsx

```diff
@@ -0,0 +1,190 @@
+
+import { usingExternalWallet } from 'common/store/pages/signon/actions'
+import { getRouteOnCompletion } from 'common/store/pages/signon/selectors'
+import { useNavigateToPage } from 'hooks/useNavigateToPage'
+import { usePortal } from 'hooks/usePortal'
+import { initSdk } from 'services/audius-sdk'
+import { audiusChain, wagmiConfig } from 'services/audius-sdk/wagmi'
+
+const messages = {
+  title: 'Are You Sure?',
+  body: 'Creating an Audius account with an external wallet will negatively impact your Audius experience in a significant way. We strongly suggest creating your account with an email and password.',
+
+  confirm: 'Yes, I Understand',
+  decline: 'No, Take Me Back',
+
+  error: 'Something went wrong. Please try again.'
+}
+
+export const doesUserExist = async (sdk: AudiusSdk, wallet: string) => {
+  try {
+    const { data } = await sdk.full.users.getUserAccount({
+      wallet
+    })
+    if (data?.user) {
+      return true
+    }
+  } catch (e) {
+    // We expect an unauthorized response for non-users
+    if (!isResponseError(e) || e.response.status !== 401) {
+      throw e
+    }
+  }
+  return false
+}
+
+export const ExternalWalletSignUpModal = () => {
+  const { isOpen, onClose } = useExternalWalletSignUpModal()
+
+  const { connectAsync } = useConnect()
+  const { switchChainAsync } = useSwitchChain()
+  const { isConnected, address } = useAccount()
+  const navigate = useNavigateToPage()
+  const dispatch = useDispatch()
+  const [status, setStatus] = useState(Status.IDLE)
+  const route = useSelector(getRouteOnCompletion)
+
+  const handleConfirm = useCallback(async () => {
+    setStatus(Status.LOADING)
+    try {
+      let wallet = address
+      // Ensure the wallet is connected
+      if (!isConnected) {
+        const connection = await connectAsync({
+          chainId: audiusChain.id,
+          connector: metaMask()
+        })
+        wallet = connection.accounts[0]
+      }
+      if (!wallet) {
+        throw new Error('No wallet connected')
+      }
+
+      // Ensure we're on the Audius chain
+      await switchChainAsync({ chainId: audiusChain.id })
+
+      // For some reason, a delay is needed to ensure the chain ID switches.
+      // I couldn't find the part of the wagmiConfig state that changes.
+      // This is partially voodoo magic, a setTimeout would also work here.
+      // Without this delay, new users without the network added will get a ConnectorChainMismatchError
+      let unsubscribe = () => {}
+      await new Promise<void>((resolve) => {
+        unsubscribe = wagmiConfig.subscribe(
+          (state) => state,
+          () => {
+            resolve()
+          }
+        )
+      })
+      unsubscribe()
+
+      // Reinit SDK with the connected wallet
+      const sdk = await initSdk()
+
+      // Check that the user doesn't already exist.
+      // If they do, log them in.
+      const userExists = await doesUserExist(sdk, wallet)
```

Comment:
previously the flow would let you go all the way until you tried to create an account and then blurt out "Something went wrong"